lib/algora/pipeline/storage/thumbnails.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Algora.Pipeline.Storage.Thumbnails do
+  @moduledoc false
+
+  require Membrane.Logger
+  alias Algora.Library
+
+  @thumbnail_markers [
+    %{minutes: 0, segment_sn: 0},
+    %{minutes: 1, segment_sn: 6},
+    %{minutes: 2, segment_sn: 14},
+    %{minutes: 4, segment_sn: 28},
+    %{minutes: 8, segment_sn: 57},
+    %{minutes: 16, segment_sn: 120}
+  ]
+
+  @pubsub Algora.PubSub
+
+  def store_thumbnail(video, video_header, contents, marker) do
+    with {:ok, video} <- Library.store_thumbnail(video, video_header <> contents, marker),
+         {:ok, video} <- Library.store_og_image(video, marker) do
+      broadcast_thumbnails_generated!(video)
+    else
+      _ ->
+        Membrane.Logger.error("Could not generate thumbnails for video #{video.id}")
+    end
+  end
+
+  def find_marker(segment_sn) do
+    Enum.find(@thumbnail_markers, fn marker ->
+      marker.segment_sn == segment_sn
+    end)
+  end
+
+  def is_last_marker?(marker) do
+    List.last(@thumbnail_markers) == marker
+  end
+
+  def mintues_for_video(video) do
+    elapsed_minutes = floor(video.duration / 60)
+    @thumbnail_markers
+    |> Enum.filter(fn marker ->
+      marker.minutes <= elapsed_minutes
+    end)
+    |> Enum.map(& &1.minutes)
+  end
```

Comment:
List of thumbnails which are probably present for a given video (based on its duration)

Reply:
With this implementation the streamer can't select a thumbnail before the stream is over though, because the duration is not updated until then

How about we persist all thumbnails into a new `video_thumbnails` table so we always know exactly which thumbnails are available for a given video, along with their timestamps?

[Check out the Ecto docs](https://hexdocs.pm/ecto_sql/Ecto.Migration.html) if you're not yet familiar with Ecto migrations

Reply:
Have implemented this. Half the implementation is upstream in (https://github.com/algora-io/tv/pull/106)